mget.go

```diff
@@ -17,36 +17,59 @@ import (
 //	    Add("example_index", "totally_real_id").
 //	    Do(context.background(), client)
 type MGetRequest struct {
-    // Index destination for entire request
-    // if used individual documents don't need to specify the index
-    Index string `json:"-"`
+	// Index destination for entire request
+	// if used individual documents don't need to specify the index
+	Index string
 
-    // Docs are the list of documents to be fetched
-    Docs []mgetDoc `json:"docs"`
+	// Docs are the list of documents to be fetched.
+	Docs []RoutableDoc
 }
 
 // NewMGetRequest instantiates an empty MGetRequest.
 // An empty MGetRequest is executable but will return zero documents because zero have been requested.
 func NewMGetRequest() *MGetRequest {
-    return &MGetRequest{}
+	return &MGetRequest{}
 }
 
-// Add a request for a document with id at a specific index.
-// If index is an empty string, the request relies on the top level index MGetRequest.Index
+// Add a DocumentRef to the documents being requested.
+// If index is an empty string, the request relies on the top level MGetRequest.Index.
 func (m *MGetRequest) Add(index, id string) *MGetRequest {
-    m.Docs = append(m.Docs, mgetDoc{
-        ID:    id,
-        Index: index,
-    })
+	return m.AddDocs(NewDocumentRef(index, id))
+}
 
-    return m
+// AddDocs - add any number RoutableDoc to the documents being requested.
+// If the doc does not return anything for [RoutableDoc.Index], the request relies on the top level MGetRequest.Index.
+func (m *MGetRequest) AddDocs(docs ...RoutableDoc) *MGetRequest {
+	m.Docs = append(m.Docs, docs...)
+	return m
 }
 
 // SetIndex sets the top level index for the request. If a individual document request does not have an index specified,
 // this index will be used.
 func (m *MGetRequest) SetIndex(index string) *MGetRequest {
-    m.Index = index
-    return m
+	m.Index = index
+	return m
+}
+
+// Source translates the MGetRequest into the shape expected by OpenSearch.
+func (m *MGetRequest) Source() map[string]any {
+	docs := make([]any, len(m.Docs))
+	for i, d := range m.Docs {
+		docReq := make(map[string]any)
```

Comment:
I think this would read simpler without the `make()` calls
```
docReq := map[string]any{
   "_id": d.ID(),
}
if d.Index() != "" {
   docReq["_index"] = d.Index()
}
docs[i] = docReq
```
same for initializing `source` below.

also, `docs[i] = d` should be `docs[i] = docReq` right?